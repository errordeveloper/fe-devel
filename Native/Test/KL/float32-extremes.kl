/*
 *  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
 */

operator entry()
{
  Scalar scalars[];
  scalars.push( 0.0 );
  scalars.push( -0.0 );
  scalars.push( 1.0 );
  scalars.push( -1.0 );

  for ( Size i=0; i<scalars.size; ++i )
  {
    for ( Size j=0; j<scalars.size; ++j )
    {
      report(scalars[i]+" * "+scalars[j]+" = "+(scalars[i]*scalars[j]));
      report(scalars[i]+" / "+scalars[j]+" = "+(scalars[i]/scalars[j]));
    }
  }
}
