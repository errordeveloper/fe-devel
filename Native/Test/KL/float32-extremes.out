0 * 0 = 0
0 / 0 = NaN
0 * -0 = -0
0 / -0 = NaN
0 * 1 = 0
0 / 1 = 0
0 * -1 = -0
0 / -1 = -0
-0 * 0 = -0
-0 / 0 = NaN
-0 * -0 = 0
-0 / -0 = NaN
-0 * 1 = -0
-0 / 1 = -0
-0 * -1 = 0
-0 / -1 = 0
1 * 0 = 0
1 / 0 = Inf
1 * -0 = -0
1 / -0 = -Inf
1 * 1 = 1
1 / 1 = 1
1 * -1 = -1
1 / -1 = -1
-1 * 0 = -0
-1 / 0 = -Inf
-1 * -0 = 0
-1 / -0 = Inf
-1 * 1 = -1
-1 / 1 = -1
-1 * -1 = 1
-1 / -1 = 1
